core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -94,19 +104,60 @@ object RequestChannel extends Logging {
     @volatile var recordNetworkThreadTimeCallback: Option[Long => Unit] = None
 
     val session = Session(context.principal, context.clientAddress)
+
     private val bodyAndSize: RequestAndSize = context.parseRequest(buffer)
 
     def header: RequestHeader = context.header
     def sizeOfBodyInBytes: Int = bodyAndSize.size
 
-    //most request types are parsed entirely into objects at this point. for those we can release the underlying buffer.
-    //some (like produce, or any time the schema contains fields of types BYTES or NULLABLE_BYTES) retain a reference
-    //to the buffer. for those requests we cannot release the buffer early, but only when request processing is done.
+    def buildResponse(abstractResponse: AbstractResponse,
+                      error: Errors): Send =
```

Comment:
nit: add braces to all of these methods. Even though they are not required, braces make it easier to see the scope